pkg/status/status.go

```diff
@@ -0,0 +1,112 @@
+package status
+
+import (
+	"context"
+	"time"
+
+	ingressnodefwv1alpha1 "github.com/openshift/ingress-node-firewall/api/v1alpha1"
+
+	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// IngressNodeFirewallConfigResourcesNotReadyError contains Error message
+// explaining the reason why ingress node firewall Config is not ready.
+type IngressNodeFirewallConfigResourcesNotReadyError struct {
+	Message string
+}
+
+func (e IngressNodeFirewallConfigResourcesNotReadyError) Error() string { return e.Message }
+
+func (e IngressNodeFirewallConfigResourcesNotReadyError) Is(target error) bool {
+	_, ok := target.(*IngressNodeFirewallConfigResourcesNotReadyError)
+	return ok
+}
+
+const (
+	// ConditionAvailable means daemonSet is available and the number of scheduled pods
+	// is the same of the number of desired pods.
+	ConditionAvailable = "Available"
+	// ConditionProgressing means daemonSet is progressing. Progress for a daemonSet
+	// is considered when desired pods are being created.
+	ConditionProgressing = "Progressing"
+	// ConditionDegraded means daemonset availability conditions not been achieved, when
+	// one of its pods fails to be created or deleted.
+	ConditionDegraded = "Degraded"
+)
+
+// Update updates ingress node firewall config object status field.
+func Update(ctx context.Context, client k8sclient.Client, infcfg *ingressnodefwv1alpha1.IngressNodeFirewallConfig, condition string, reason string, message string) error {
+	conditions := getConditions(condition, reason, message)
+	if equality.Semantic.DeepEqual(conditions, infcfg.Status.Conditions) {
+		return nil
+	}
+	infcfg.Status.Conditions = getConditions(condition, reason, message)
```

Comment:
mistake right? just set `conditions` var.

Reply:
yeah no need to call getConditions again